presidio-analyzer/analyzer/recognizer_registry/recognizer_registry.py

```diff
@@ -0,0 +1,65 @@
+from analyzer import PatternRecognizer, RemoteRecognizer
+from analyzer.predefined_recognizers import CreditCardRecognizer, SpacyRecognizer, CryptoRecognizer, DomainRecognizer, EmailRecognizer, IbanRecognizer, IpRecognizer, NhsRecognizer, UsBankRecognizer, UsLicenseRecognizer, UsItinRecognizer, UsPassportRecognizer, UsPhoneRecognizer, UsSsnRecognizer
+
+
+class RecognizerRegistry:
+    """
+    Detects, registers and holds all recognizers to be used by the analyzer
+    """
+
+    def __init__(self, recognizers=None):
+        if recognizers is None:
+            recognizers = []
+        self.recognizers = recognizers
+
+    def load_pattern_recognizer(self, data):
+        """
+        Load pattern recognizer from json file
+        :param data: a dictionary (possibly created from a json) which holds the parameters of a recognizer
+        """
+        self.recognizers.append(PatternRecognizer.from_dict(data))
+
+    def load_external_recognizer(self, data):
+        '''
+        Load external recognizer (in separate container) from json metadata file
```

Comment:
container as in docker container? if so not sure this code should 'know' how it runs, that it have containers